core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java

```diff
@@ -512,24 +513,28 @@ public List<String> getManagerLocations() {
     OpTimer timer = null;
 
     if (log.isTraceEnabled()) {
-      log.trace("tid={} Looking up manager location in zookeeper.", Thread.currentThread().getId());
+      log.trace("tid={} Looking up manager location in zookeeper at {}.",
+          Thread.currentThread().getId(), zLockManagerPath);
       timer = new OpTimer().start();
     }
 
-    byte[] loc = zooCache.getLockData(zLockManagerPath);
+    ServiceLockData sld = zooCache.getLockData(zLockManagerPath);
+    String location = null;
+    if (sld != null) {
+      location = sld.getAddressString(ThriftService.MANAGER);
+    }
```

Comment:
You could make it so `sld` is never `null`, which would clean up some of these `null` checks, which seems to be a lot of boilerplate. You could return `Optional<ServiceLockData>` or a dummy `ServiceLockData` that represents the "not found in ZK" case (like `ServiceLockData.MISSING` that has a `null` address string).

Reply:
Updated to use Optional